### ISSUE TITLE
Fedora (linux) : trouble building the latest master

### ISSUE THREAD

**Issue by 7oxicshadow** [RPCS3/rpcs3#2852]
When trying to get RPCS3 to build on Fedora I am currently experiencing 2 issues:

First is not a major issue but i get the following warning:

```
CMake Warning at rpcs3/CMakeLists.txt:114 (find_package):
  Could not find a configuration file for package "LLVM" that is compatible
  with requested version "4.0".

  The following configuration files were considered but not accepted:

    /usr/lib64/cmake/llvm/LLVMConfig.cmake, version: 3.9.1
    /lib64/cmake/llvm/LLVMConfig.cmake, version: 3.9.1

-- RPCS3: using builtin libpng
-- RPCS3: using builtin ffmpeg
LLVM not found! LLVM 4.0 is required. RPCS3 will be compiled without LLVM support.
```
I have all of the submodules from git and LLVM is amongst them. Is there a way to force RPCS3 to use the submodule version?

The second issue fails at the linker stage:

```
[100%] Linking CXX executable ../bin/rpcs3
/usr/bin/ld: cannot find /usr/lib64/libatomic.so.1.2.0
```

I have libatomic installed but I cannot find any reference to libatomic-devel within DNF. Are there any other Fedora users that have succeeded in building the very latest git revision. I am struggling to find which package provides the required file(s)?

Thanks.

**Comment by hcorion** [comment]
The submodule is for Windows users, so they can build llvm.
Fedora it seems hasn't updated to 4.0 yet (understandable).
If you really want LLVM with RPCS3, you can build and install it via the submodule.
LLVM build guide: http://llvm.org/docs/CMake.html

---

Regarding the issue with libatomic, GCC 7 now requires manually linking with `-latomic`, what GCC version are you using? Perhaps we'll have to only link with `-latomic` with GCC 7 or greater.

**Comment by 7oxicshadow** [comment]
Thanks for your reply. I have tried building with both GCC and CMAKE. Both fail with the libatomic problem.

gcc version 6.3.1 20161221 (Red Hat 6.3.1-1) (GCC)
cmake version 3.8.0

**Comment by 7oxicshadow** [comment]
I have removed -latomic from line 284 of rpcs3/rpcs3/CMakeLists.txt and the build completes successfully with no errors.

I think your GCC 7 only suggestion is correct.

**Comment by AniLeo** [comment]
Fixed by #2867
